src/Libs/OwlCore.WinUI/Behaviors/OpenUriAction.cs

```diff
@@ -9,16 +9,16 @@ namespace OwlCore.WinUI.Behaviors
     /// <summary>
     /// A behavior action that opens the provided parameter as a Uri.
     /// </summary>
-    public class OpenUriAction : DependencyObject, IAction
+    public partial class OpenUriAction : DependencyObject, IAction
     {
-        public Uri? Source { get; set; }
+        public string? Source { get; set; }
 
         /// <inheritdoc />
         public object? Execute(object sender, object parameter)
         {
             Guard.IsNotNull(Source);
 
-            _ = Launcher.LaunchUriAsync(Source);
+            _ = Launcher.LaunchUriAsync(new Uri(Source));
```

Comment:
Hmm why not await it? does it block the execution, I think it also just delegate the heavy lifting to some other thread and returns the operation states pretty quickly.

Reply:
Behavior handlers are fire-and-forget and cannot do async. Even if we could, we don't need to make an async state machine to open a URL.